CRM/Contact/BAO/ContactType.php

```diff
@@ -53,48 +55,24 @@ public static function isActive($contactType) {
   /**
    * Retrieve basic contact type information.
    *
-   * @param bool $all
+   * @param bool $includeInactive
    *
    * @return array
    *   Array of basic contact types information.
+   *
+   * @throws \API_Exception
+   * @throws \Civi\API\Exception\UnauthorizedException
    */
-  public static function basicTypeInfo($all = FALSE) {
-    static $_cache = NULL;
-
-    if ($_cache === NULL) {
-      $_cache = [];
-    }
-
-    $argString = $all ? 'CRM_CT_BTI_1' : 'CRM_CT_BTI_0';
-    if (!array_key_exists($argString, $_cache)) {
-      $cache = CRM_Utils_Cache::singleton();
-      $_cache[$argString] = $cache->get($argString);
-      if (!$_cache[$argString]) {
-        $sql = "
-SELECT *
-FROM   civicrm_contact_type
-WHERE  parent_id IS NULL
-";
-        if ($all === FALSE) {
-          $sql .= " AND is_active = 1";
-        }
-
-        $params = [];
-        $dao = CRM_Core_DAO::executeQuery($sql,
-          $params,
-          FALSE,
-          'CRM_Contact_DAO_ContactType'
-        );
-        while ($dao->fetch()) {
-          $value = [];
-          CRM_Core_DAO::storeValues($dao, $value);
-          $_cache[$argString][$dao->name] = $value;
-        }
-
-        $cache->set($argString, $_cache[$argString]);
+  public static function basicTypeInfo($includeInactive = FALSE) {
+    $cacheKey = 'CRM_CT_BTI_' . (bool) $includeInactive;
```

Comment:
I was curious about this because concatenating a boolean onto a string sounds like it might produce unexpected results.

```cli
> cv ev 'return "hello" . (bool) true;'
"hello1"
> cv ev 'return "hello" . (bool) false;'
"hello"
```

Turns out the results *are* different, and consistently so, and that's the only requirement in this instance (as long as PHP doesn't change its policy on this kinda thing in the future) but as a matter of personal preference I'd probably cast it to `(int)`. Or I guess if you want to get really pick the way to make it always "1" or "0" would be:
```suggestion
    $cacheKey = 'CRM_CT_BTI_' . (int) (bool) $includeInactive;
```

Reply:
@colemanw ok - cast to int